samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java

```diff
@@ -265,15 +268,22 @@ protected Object getEnvelopePartitionId(OutgoingMessageEnvelope envelope) {
   }
 
   private String convertPartitionKeyToString(Object partitionKey) {
+    String ret;
     if (partitionKey instanceof String) {
-      return (String) partitionKey;
+      ret = (String) partitionKey;
     } else if (partitionKey instanceof Integer) {
-      return String.valueOf(partitionKey);
+      ret = String.valueOf(partitionKey);
     } else if (partitionKey instanceof byte[]) {
-      return new String((byte[]) partitionKey, Charset.defaultCharset());
+      ret = new String((byte[]) partitionKey, Charset.defaultCharset());
     } else {
       throw new SamzaException("Unsupported key type: " + partitionKey.getClass().toString());
     }
+    if (ret != null && ret.length() > ClientConstants.MAX_PARTITION_KEY_LENGTH) {
+      LOG.debug("length of partitioning key ({}) exceeds limit: {}. Will truncate the key...", ret.length(),
```

Comment:
"Length of partition key: foo exceeds limit: 5000 bytes. Truncating"

Reply:
1. Additionally would be nice to have a metric for truncated message count. 
2. Consider adding an unit-test for this as well if it's not too much work.